packages/react/src/transactions/adapters/PublicationCacheManager.ts

```diff
@@ -0,0 +1,145 @@
+import { ApolloCache, DocumentNode, NormalizedCacheObject } from '@apollo/client';
+import {
+  CommentFragmentDoc,
+  JustTypename,
+  MirrorFragmentDoc,
+  PostFragmentDoc,
+  PublicationFragment,
+} from '@lens-protocol/api-bindings';
+import { failure, invariant, never, Result, success } from '@lens-protocol/shared-kernel';
+
+export class FragmentNotFoundError extends Error {
+  name = 'FragmentNotFoundError' as const;
+}
+
+export const getFragmentDoc = (publication: JustTypename<PublicationFragment>): DocumentNode => {
+  switch (publication.__typename) {
+    case 'Mirror':
+      return MirrorFragmentDoc;
+    case 'Post':
+      return PostFragmentDoc;
+    case 'Comment':
+      return CommentFragmentDoc;
+    default:
+      never("Can't infer fragment document");
+  }
+};
+
+export class PublicationCacheManager {
```

Comment:
@cesarenaldi @krzysu @reecejohnson @JoaquinBattilana let me know your thoughts about the new `PublicationCacheManager`.

The overall idea is to abstract from presenters/responders/unit-tests the exact way to update the cache and keep the knowledge of `how` in a single place.

On top of that `PublicationCacheManager` abstracts away the knowledge of knowing exactly which fragment to update in case of a publication (which can be either `PostFragment`,`CommentFragment`,`MirrorFragment`). Given that seems we can get rid of `PublicationType` from the `@lens-protocol/domain` package once we migrate presenters to use `PublicationCacheManager`.

Reply:
i like the centralised place to update all publication cache, good idea